packages/react-graphql/src/hooks/query.ts

```diff
@@ -1,10 +1,12 @@
+import {useEffect, useMemo, useState, useRef} from 'react';
 import {
   OperationVariables,
   ApolloError,
   WatchQueryOptions,
 } from 'apollo-client';
 import {DocumentNode} from 'graphql-typed';
-import {useEffect, useMemo, useState, useRef} from 'react';
+import {useServerEffect} from '@shopify/react-effect';
+import {useAsyncAsset} from '@shopify/react-async';
```

Comment:
`QueryHookOptions` does include a variable call `ssr`, 
and there are some special case the [`react-apollo-hooks`](https://github.com/trojanowski/react-apollo-hooks/blob/master/src/useQuery.ts#L86-L87) does do and I didn't take into account originally.

should `useServerEffect` &  `useAsyncAsset` always applied or should it be dependent on the `ssr` variable & has it affect the `fetchPolicy`?